target/linux/generic/files/drivers/net/phy/rtl8366_smi.c

```diff
@@ -1550,6 +1550,12 @@ int rtl8366_smi_probe_of(struct platform_device *pdev, struct rtl8366_smi *smi)
 		goto try_gpio;
 	}
 
+	if (of_property_read_u32(np, "phy_id", &smi->phy_id) < 0) {
+		smi->phy_id = MDC_REALTEK_PHY_ADDR;
+	}
+
```

Comment:
Oh sorry I meant IS_ERR_VALUE()...
```
if (IS_ERR_VALUE(of_property_read_u32(np, "phy_id", &smi->phy_id)))
		smi->phy_id = MDC_REALTEK_PHY_ADDR;

```

Reply:
Sorry again :flushed:. I don't know what's wrong with me today... of_property_read_u32() never returns positive values > 0.
Thus it's just:
```
if (of_property_read_u32(np, "phy_id", &smi->phy_id))
	smi->phy_id = MDC_REALTEK_PHY_ADDR;
```